front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -227,70 +265,90 @@ const SpaceTimeChartWrapper = ({
           loading,
-        }) => ({
-          id: operationalPointId,
-          position: operationalPointPosition,
-          size: (tracks?.length || 0) * TRACK_HEIGHT_CONTAINER + DEFAULT_THEME.timeCaptionsSize,
-          spaceTimeChartNode: (
-            <TrackOccupancyCanvas
-              position={operationalPointPosition}
-              tracks={tracks || []}
-              occupancyZones={(zones || []).map((zone) => {
-                const path = pathsIndex[zone.trainId];
-                if (!path) return zone;
-                const pathStyle = getPathStyle(
-                  hoveredItem,
-                  path,
-                  !!draggingState,
-                  subCategories,
-                  timetableItemsWithDetails,
-                  selectedTrainId
-                );
-                return {
-                  ...zone,
-                  color: pathStyle.color,
-                  size: pathStyle.level === 1 ? 2 : undefined,
-                };
-              })}
-              selectedTrainId={selectedTrainId}
-              onClose={() => onCloseOccupancyLayer?.(waypointId)}
-              topPadding={BASE_WAYPOINT_HEIGHT}
-            />
-          ),
-          manchetteNode: (
-            <TrackOccupancyManchette tracks={tracks || []}>
-              <div className="waypoint-wrapper flex justify-start">
-                <WaypointComponent
-                  waypoint={{
-                    id: waypointId,
-                    name: (
-                      <div className="d-flex flex-row align-items-center">
-                        {operationalPointName || operationalPointId}
-                        {loading && (
-                          <Spinner className="ml-2 small" spinnerClassName="spinner-border-sm" />
-                        )}
-                      </div>
-                    ),
-                    position: operationalPointPosition,
-                    onClick: handleWaypointClick,
-                  }}
-                  waypointRef={activeWaypointRef}
-                  isActive={false}
-                  isMenuActive={false}
-                />
-              </div>
-            </TrackOccupancyManchette>
-          ),
-        })
-      ) || []
-    );
-  }, [
-    trackOccupancyDiagramsData,
-    activeWaypointId,
-    timetableItemsWithDetails,
-    selectedTrainId,
-    hoveredItem,
-  ]);
+        }) => {
+          const baseZones = zones ?? [];
+          const zonesCountByPacedTrainId = countZonesByPacedTrainId(baseZones);
+
+          const styledZones = baseZones.map((zone) => {
+            const path = pathsById[zone.trainId];
+
+            const isHovered = hoveredPathId === zone.trainId;
+            const isSelected = selectedTrainId === zone.trainId;
+
+            const paced = toPaced(zone.trainId);
+            const totalOnTrack =
+              paced && zone.trackId
+                ? (zonesCountByPacedTrainId.get(paced)?.get(zone.trackId) ?? 0)
+                : 0;
```

Comment:
nit: I find it easier to read that you are handling a paced train. Also you drop the function `toPaced`

Note: `trainId` is an `TrainId` (so `OccurrenceId` or `TrainScheduleId`)
```suggestion
            let totalOccurrencesOnTrack = 0;
            if (isOccurrenceId(zone.trainId)) {
              const pacedTrainId = extractPacedTrainIdFromOccurrenceId(zone.trainId);
              totalOccurrencesOnTrack =
                zonesCountByPacedTrainId.get(pacedTrainId)?.get(zone.trackId) ?? 0;
            }
```